### ISSUE TITLE
VPA support for pod-level resource specifications

### ISSUE THREAD

**Issue by voelzmo** [kubernetes/autoscaler#7571]
**Which component are you using?**:
/area vertical-pod-autoscaler

**Is your feature request designed to solve a problem? If so describe the problem this feature should solve.**:
With k8s 1.32, a new feature gate is released in alpha: `PodLevelResources`, allowing users to specify resources on Pod level instead of on Container level.
More info on how this works in the [KEP](https://github.com/kubernetes/enhancements/issues/2837)

**Describe the solution you'd like.**:
VPA should support this and autoscale Pods instead of individual Containers.
This is just a placeholder/umbrella issue, we need a proper enhancement proposal doc for this.

**Comment by omerap12** [comment]
I’d be happy to open a KEP for this, as I have some ideas to contribute. Specifically, I believe the VPA should support both approaches. That means, if we have PodLevelResources, the scaling should apply at the pod level; otherwise, it should scale at the container level. The main challenge lies in scaling multi-container pods with PodLevelResources. I shared an idea on this topic here: [comment](https://github.com/kubernetes/autoscaler/issues/7147#issuecomment-2515296024).

**Comment by omerap12** [comment]
This issue still exists and needs to be addressed. I don't currently have the capacity to work on it, so any contributions would be greatly appreciated.
/unassign
/help
/open

**Comment by iamzili** [comment]
@omerap12

I can work on this issue. BTW based on the comments, a proposal (KEP) needs to be created first, right?

**Comment by omerap12** [comment]
> @omerap12
> 
> I can work on this issue. BTW based on the comments, a proposal (KEP) needs to be created first, right?
> 
> 
> 

Cool. Yeah an AEP would be ideal :)

**Comment by adrianmoisey** [comment]
@iamzili just checking if you're still working on this? Shout if you need a hand

**Comment by iamzili** [comment]
@adrianmoisey yep, I'm still working on it. In a couple of days I will finish the AEP.

**Comment by iamzili** [comment]
fyi @adrianmoisey and @omerap12 

I created an initial version of the KEP, which discusses pod-level `resources` support in VPA. It is still a work in progress, as the community needs to agree on the default behavior of VPA based on the types of resource stanzas defined  (e.g. when a user deploys a Deployment that specifies both pod and container-level resources):

[KEP](https://github.com/kubernetes/autoscaler/pull/8586)
